tscreen.go

```diff
@@ -1043,6 +1043,14 @@ func (t *tScreen) enablePasting(on bool) {
 	}
 }
 
+func (t *tScreen) enableFocusReporting() {
+	t.TPuts("\x1b[?1004h")
```

Comment:
This is the part of your code I have a problem with.

Not every terminal will understand these, and for those that don't this may be problematic.

At a minimum this probably needs to be limited to only being used for terminals that support mouse reporting.  I *think* probably any terminal that supports mouse reporting will also support this.

Also, some terminals may want to override this value.  See the enablePaste and disablePaste terminal settings.

FIxing this to do it the right way might be more effort than you want to undertake.  If it is, let me know, and I can probably take this the rest of the way over the finish line -- this does seem like it could well be a useful feature and so I want to have it included.  (And thank you very much for taking it this far already, btw.)

Reply:
I pushed a simple attempt at doing this (see 7521492e80db), but I'm not sure I understand all of it. For `Terminfo.EnablePaste` I don't see any code anywhere that actually sets it to something...

Reply:
I commented there... have a look at how the prepareCursorStyles() or prepareBracketedPaste works.

Reply:
maybe add stuff to prepareExtendedOSC ?  This isn't an OSC, but looking at it, neither is the code for setting Window Size.

Reply:
Is this getting closer? 0732b364fb37